packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -5032,6 +5032,8 @@ class Column extends Flex {
 
 /// A widget that controls how a child of a [Row], [Column], or [Flex] flexes.
 ///
+/// This widget can be place inside a [PreferredSize] if it respect the above rule.
```

Comment:
Hi @EArminjon, I am not sure what this means. Can you elaborate?

Reply:
Hi @Piinks , as a developer I didn't know that it was possible to use Expanded or Flexible under a PreferredSize widget. The doc didn't mention that. Many of my co-workers think that Expanded and Flexible must be first widget child of Column / Row / Flex. 

I think doc need this little update.

Reply:
Thank you! I appreciate you contributing!
Can you expand a bit more on why someone might be looking for this? Or what effect they might use this for? Any widget can be placed within a PreferredSize widget, why in particular would this case be important to call out?
Is there some sample code that illustrates this maybe?
Or should this be documented in PreferredSize instead?

Reply:
The purpose of this PR is just to clarify that we can use Flexible and Expanded under a Preferred size widget and not only directly under a Column, Row or Flex widget. Me and my co-workers use flutter since 3 years and we just discover that today... 
If there is other possible case. It could be interesting to know they too :)

Reply:
I don't disagree with you. I just wonder if you might be able to provide more context here? As a reader, I do not know why this is helpful information. What might I want to do?

Reply:
I don't know what to add more. It's possible to do that and the flutter doc didn't mention it :/

Reply:
For me, when I read this, I don't know why I would do this. You can place a Flexible widget inside lots of different widgets. Why this one? What might the user be trying to do that this would be the answer for? It is not clear here.

To suggest it another way, how is this different or helpful compared to saying

```suggestion
/// This widget can be place inside a [Container] if it respect the above rule.
```

What is the difference?

Reply:
If you put some attribute to your container it will fail and you will see the following message  : 

```
he following assertion was thrown while applying parent data.:
Incorrect use of ParentDataWidget.

The ParentDataWidget Flexible(flex: 1) wants to apply ParentData of type FlexParentData to a RenderObject, which has been set up to accept ParentData of incompatible type BoxParentData.
```

Reply:
I agree with @Piinks that this sentence alone is not very useful and probably more confusing. If you want to include this information, can you please **update the doc** with answers to the questions @Piinks was asking above? In particular, the docs should indicate WHY one should places this widget inside a PreferredSize. What effect can you achieve with that that cannot be achieved otherwise? The docs should explicitly explain that.

FWIW, I don't think that this is correct usage of the PreferredSize widget and I am not really sure why you would want to place this widget inside a PreferredSize like that. The example given in the PR description could easily be rewritten to give the same result without the confusing usage of PreferredSize:

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(
    const MaterialApp(
      home: Home(),
    ),
  );
}

class Home extends StatelessWidget {
  const Home({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Column(
        children: [
           Expanded(
            child: Container(
              padding: const EdgeInsets.all(20),
              color: Colors.red,
              width: double.maxFinite,
            ),
          ),
          Expanded(
            child: Container(
              padding: const EdgeInsets.all(20),
              color: Colors.orange,
              width: double.maxFinite,
            ),
          ),
        ],
      ),
    );
  }
}
```

Reply:
> I agree with @Piinks that this sentence alone is not very useful and probably more confusing. If you want to include this information, can you please **update the doc** with answers to the questions @Piinks was asking above? In particular, the docs should indicate WHY one should places this widget inside a PreferredSize. What effect can you achieve with that that cannot be achieved otherwise? The docs should explicitly explain that.
> 
> FWIW, I don't think that this is correct usage of the PreferredSize widget and I am not really sure why you would want to place this widget inside a PreferredSize like that. The example given in the PR description could easily be rewritten to give the same result without the confusing usage of PreferredSize:
> 
> ```dart
> import 'package:flutter/material.dart';
> 
> void main() {
>   runApp(
>     const MaterialApp(
>       home: Home(),
>     ),
>   );
> }
> 
> class Home extends StatelessWidget {
>   const Home({super.key});
> 
>   @override
>   Widget build(BuildContext context) {
>     return Scaffold(
>       body: Column(
>         children: [
>            Expanded(
>             child: Container(
>               padding: const EdgeInsets.all(20),
>               color: Colors.red,
>               width: double.maxFinite,
>             ),
>           ),
>           Expanded(
>             child: Container(
>               padding: const EdgeInsets.all(20),
>               color: Colors.orange,
>               width: double.maxFinite,
>             ),
>           ),
>         ],
>       ),
>     );
>   }
> }
> ```

The code example is for example only. Not real usage.

Reply:
This intent of this PR is just to show a possible usage. Don't have precise example in mind. It's working and was not mentioned that's why :/